### ISSUE TITLE
history json OSError bad file descriptor

### ISSUE THREAD

**Issue by laerus** [xonsh/xonsh#2122]
occasionally on xonsh startup:
```
Exception in thread Thread-5:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/home/laerus/src/xonsh/xonsh/history/json.py", line 103, in run
    files = self.files(only_unlocked=True)
  File "/home/laerus/src/xonsh/xonsh/history/json.py", line 126, in files
    fs = _xhj_get_history_files(sort=False)
  File "/home/laerus/src/xonsh/xonsh/history/json.py", line 70, in _xhj_get_history_files
    files = [os.path.join(data_dir, f) for f in os.listdir(data_dir)
OSError: [Errno 9] Bad file descriptor: '/home/laerus/.local/share/xonsh'
```

**Comment by scopatz** [comment]
Hmm Odd. Thanks for reporting @laerus. I assume that the directory actually exists?

**Comment by laerus** [comment]
haven't deleted anything manually for some time, only reinstalled from master

**Comment by scopatz** [comment]
It would be nice if we could find a way to reliably reproduce this. We can always put a try-except around this, but it would be nice to know what is going wrong.

**Comment by scopatz** [comment]
See safety fix in #2284

**Comment by melund** [comment]
#2284 is merged. If anyone sees the issue again feel free to reopen.
